playbooks/robusta_playbooks/node_actions.py

```diff
@@ -0,0 +1,76 @@
+from typing import List
+
+from kubernetes import client
+from kubernetes.client import V1OwnerReference, V1PodList
+
+from robusta.api import ActionException, ErrorCodes, MarkdownBlock, NodeEvent, action
+
+
+@action
+def cordon(event: NodeEvent):
+    """
+    Cordon a node
+    """
+    node = event.get_node()
+    if not node:
+        raise ActionException(ErrorCodes.RESOURCE_NOT_FOUND, "Failed to get node for cordon")
+
+    if node.spec.unschedulable:
+        event.add_enrichment([MarkdownBlock(f"Node {node.metadata.name} already cordoned")])
+        return
+
+    try:
+        client.CoreV1Api().patch_node(node.metadata.name, {"spec": {"unschedulable": True}})
+        event.add_enrichment([MarkdownBlock(f"Node {node.metadata.name} cordoned")])
+    except Exception as e:
+        raise ActionException(ErrorCodes.ACTION_UNEXPECTED_ERROR, f"Failed to cordon node {node.metadata.name} {e}")
+
+
+@action
+def uncordon(event: NodeEvent):
+    """
+    Unordon a node
+    """
+    node = event.get_node()
+    if not node:
+        raise ActionException(ErrorCodes.RESOURCE_NOT_FOUND, "Failed to get node for uncordon")
+
+    if node.spec.unschedulable is False:
+        event.add_enrichment([MarkdownBlock(f"Node {node.metadata.name} already uncordoned")])
+        return
+
+    try:
+        client.CoreV1Api().patch_node(node.metadata.name, {"spec": {"unschedulable": False}})
+        event.add_enrichment([MarkdownBlock(f"Node {node.metadata.name} uncordoned")])
+    except Exception as e:
+        raise ActionException(ErrorCodes.ACTION_UNEXPECTED_ERROR, f"Failed to uncordoned node {node.metadata.name} {e}")
+
+
+@action
+def drain(event: NodeEvent):
```

Comment:
I would add a comment about the function
also I would add these to the docs or talk to pavan about this

Reply:
this function leaves the node in a cordoned state, so no new pods will be scheduled, i would add this to the comment

Reply:
also how long does this take on a fully loaded node? just an estimate?

Reply:
I think the action is auto generated, will double check

Reply:
also takes like few seconds for 150 pods